OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -388,6 +389,10 @@ abstract class MovePart : Activity
 
 				IsInterruptible = false; // See comments in Move.Cancel()
 
+				TurnsWhileMoving = move.mobile.Info.TurnsWhileMoving;
+				if (TurnsWhileMoving)
+					ToFacing = TurnsWhileMoving ? (To - From).Yaw + Move.ActorFacingModifier : toFacing;
```

Comment:
this is better, but as I mentioned in https://github.com/OpenRA/OpenRA/pull/20794/files#r1160623352 it would be even better to move this to where MovePart is instantiated. Then we could further optimise / make the code cleaner by merging behaviours. 

For instance in Move `new MoveFirstHalf` you only need to pass `firstFacing`, as that is already the correct facing. Only in MoveFirstHalf `new MoveSecondHalf` you need to add this calculation

Reply:
Or something along those lines. I think my suggestion had a bug

Reply:
Tried, it is sad that it is too good to be true:
![bug](https://user-images.githubusercontent.com/13763394/230718523-42df6c37-1961-4fee-896c-b239e6831291.gif)
https://github.com/dnqbob/OpenRA/commit/6293feb51a066cc5770ced953df2b618c30f41cc

The turning looks stiff, and sometimes looks too sudden. I guess it is due to the `delta`
![捕获](https://user-images.githubusercontent.com/13763394/230718779-69fcd0a8-1717-41d0-a633-1377551e5374.PNG)

We still need to keep the `delta` to orignal value, instead of the facing from `TurnWhileMoving`.

Reply:
Yeah. That's a problem

Reply:
will look into it later

Reply:
Add your commit as [Disable the arcing when moving if TurnsWhileMoving](https://github.com/OpenRA/OpenRA/pull/20794/commits/514469d42ca8128c6944dbbc70899c255217fef8)

Reply:
I think you are better off just squashing. The commit name is both inaccurate and is just refactoring the previous commit.